pkg/apis/metrics/types.go

```diff
@@ -16,14 +16,427 @@ limitations under the License.
+	Network    NetworkStats `json:"network,omitempty"`
+	// Filesystem statistics
+	HasFilesystem bool      `json:"hasFilesystem,omitempty"`
+	Filesystem    []FsStats `json:"filesystem,omitempty"`
+	// Task load statistics
+	HasLoad bool      `json:"hasLoad,omitempty"`
+	Load    LoadStats `json:"load,omitempty"`
+	// Custom Metrics
+	HasCustomMetrics bool           `json:"hasCustomMetrics,omitempty"`
+	CustomMetrics    []CustomMetric `json:"customMetrics,omitempty"`
+}
+
+type TcpStat struct {
+	// Number of connections in the Established state.
+	Established uint64 `json:"established,omitempty"`
+	// Number of connections in the SynSent state.
+	SynSent uint64 `json:"synSent,omitempty"`
+	// Number of connections in the SynRecv state.
+	SynRecv uint64 `json:"synRecv,omitempty"`
+	// Number of connections in the FinWait1 state.
+	FinWait1 uint64 `json:"finWait1,omitempty"`
+	// Number of connections in the FinWait2 state.
+	FinWait2 uint64 `json:"finWait2,omitempty"`
+	// Number of connections in the TimeWait state.
+	TimeWait uint64 `json:"timeWait,omitempty"`
+	// Number of connections in the Close state.
+	Close uint64 `json:"close,omitempty"`
+	// Number of connections in the CloseWait state.
+	CloseWait uint64 `json:"closeWait,omitempty"`
+	// Number of connections in the LastAck state.
+	LastAck uint64 `json:"lastAck,omitempty"`
+	// Number of connections in the Listen state.
+	Listen uint64 `json:"listen,omitempty"`
+	// Number of connections in the Closing state.
+	Closing uint64 `json:"closing,omitempty"`
+}
+
+type NetworkStats struct {
+	// Network stats by interface.
+	Interfaces []InterfaceStats `json:"interfaces,omitempty"`
+	// TCP connection stats (Established, Listen...)
+	Tcp TcpStat `json:"tcp,omitempty"`
+	// TCP6 connection stats (Established, Listen...)
+	Tcp6 TcpStat `json:"tcp6,omitempty"`
+}
+
+type InterfaceStats struct {
+	// The name of the interface.
+	Name string `json:"name,omitempty"`
+	// Cumulative count of bytes received.
+	RxBytes uint64 `json:"rxBytes,omitempty"`
+	// Cumulative count of packets received.
+	RxPackets uint64 `json:"rxPackets,omitempty"`
+	// Cumulative count of receive errors encountered.
+	RxErrors uint64 `json:"rxErrors,omitempty"`
+	// Cumulative count of packets dropped while receiving.
+	RxDropped uint64 `json:"rxDropped,omitempty"`
+	// Cumulative count of bytes transmitted.
+	TxBytes uint64 `json:"txBytes,omitempty"`
+	// Cumulative count of packets transmitted.
+	TxPackets uint64 `json:"txPackets,omitempty"`
+	// Cumulative count of transmit errors encountered.
+	TxErrors uint64 `json:"txErrors,omitempty"`
+	// Cumulative count of packets dropped while transmitting.
+	TxDropped uint64 `json:"txDropped,omitempty"`
+}
+
+// Instantaneous CPU stats
+type CpuInstStats struct {
+	Usage CpuInstUsage `json:"usage,omitempty"`
+}
+
+// CPU usage time statistics.
+type CpuInstUsage struct {
+	// Total CPU usage.
+	// Units: nanocores per second
```

Comment:
Can we combine the instantaneous metric and the below struct by adding a Period field? That is, all of the values become a duration, and we add a "period"/"duration" field, so that you can just divide if you need to.

Also shouldn't this unit be something like nanocore-seconds per second?